config.schema.yaml

```diff
@@ -38,7 +36,7 @@ smtp_accounts:
       smtp_auth:
         description: "Authenticate username and password with SMTP server to send email. Default True"
         type: "boolean"
-        default: true
+        default: false
```

Comment:
Why change the default smtp_auth behaviour from true to false. The description still states default is True.
I think the other changes to make username and password optional are good, but I don't think the value of smtp_auth default should be altered. This will affect anyone who already uses the pack and relies on the fact that the default value of smtp_auth is true.

Reply:
It doesn't make much sense to set smtp_auth to true and then NOT require username/password.

Meaning if I'm someone using this pack for the first time and I don't supply a username/password (because I don't need it), but smtp_auth is set to true, then I'm going to get auth failures right out of the box.

If you're determined to have this stay to default of true, then we should put it back as a required item so it gives someone the change to set it to false.

Reply:
And yes I now realize I missed the description stating it was defaulted to true. Thanks for pointing that out.

Reply:
> It doesn't make much sense to set smtp_auth to true and then NOT require username/password.
> 
> Meaning if I'm someone using this pack for the first time and I don't supply a username/password (because I don't need it), but smtp_auth is set to true, then I'm going to get auth failures right out of the box.
> 
> If you're determined to have this stay to default of true, then we should put it back as a required item so it gives someone the change to set it to false.

I think its fine to default it to true and have as optional. We can just make sure the README makes sense. I don't think we should change the default behaviour, else anyone using it and upgrades the pack can move from using smtp_auth to not using it.
We don't have conditional logic on the required, so it would make sense for username/password to be optional - as it dpeends on what other parameters you have as to whether you specify them or not. So I think its find for username password to be optional, but keep smtp_auth default to true.